core/src/net/network_channel.rs

```diff
@@ -434,6 +441,11 @@ impl std::fmt::Debug for ChannelState {
                 .field("addr", addr)
                 .field("id", id)
                 .finish(),
+            ChannelState::Error(error) => f
+                .debug_struct("ChannelState")
```

Comment:
How about we move the `debug_struct` and `finish` parts out of the `match`? This is getting really long and there's a lot of duplication between the branches.

Also, why are the `State` and `Error` fields capitalised, but real fields lower case?